### PR TITLE
Remove .visit(index: Int) args

### PR DESCRIPTION
**Pull request by htmldoug**
I was thinking about how to hide the `index: Int` arg in examples to avoid scaring new users when I realized it's never actually used by anything. When problems happen, the low level code just throws and lets the exception get enriched with parser level info (#8, #37) at a higher level. It's pointless complexity.

The one caveat is that we'll lose index positions we've already consumed in the situation where input is buffered. Buffering currently only happens when waiting for the "$type" discriminator, but in this case, the macro exceptions tell you exactly which fields are missing. The index of positions within the object add little to no value.

### CODE REVIEW

File: weejson/src/com/rallyhealth/weejson/v1/IndexedValue.scala [hunk 1]
```diff
@@ -1,96 +1,101 @@
...
-      case IndexedValue.Obj(i, items @_*) =>
-        val ctx = f.visitObject(-1, -1).narrow
+      case IndexedValue.Null => f.visitNull()
+      case IndexedValue.True => f.visitTrue()
+      case IndexedValue.False => f.visitFalse()
+      case IndexedValue.Str(s) => f.visitString(s)
+      case IndexedValue.Num(s, d, e) => f.visitFloat64StringParts(s, d, e)
+      case IndexedValue.NumLong(l) => f.visitInt64(l)
+      case IndexedValue.NumRaw(d) => f.visitFloat64(d)
+      case IndexedValue.Timestamp(i) => f.visitTimestamp(i)
```
Comment (htmldoug) [inline weejson/src/com/rallyhealth/weejson/v1/IndexedValue.scala]: I made some additions here.

File: weejson/src/com/rallyhealth/weejson/v1/IndexedValue.scala [hunk 2]
```diff
@@ -1,96 +1,101 @@
...
+  * A version of [[com.rallyhealth.weejson.v1.Value]] used to buffer data in raw form.
   *
   * This is used by the case class macros to buffer data for polymorphic types
   * when the discriminator is not the first element, e.g. {"foo": 1, "$type": "discriminator"}.
   * It is important that all types be immutable.
   */
-sealed trait IndexedValue {
-  def index: Int
-}
+sealed trait IndexedValue
```
Comment (jducoeur) [inline weejson/src/com/rallyhealth/weejson/v1/IndexedValue.scala]: Is this still meaningful, without the index?  Is the name appropriate?

Comment (htmldoug) [reply weejson/src/com/rallyhealth/weejson/v1/IndexedValue.scala]: It's really not. Renamed to `BufferedValue`.

File: weepickle/src/com/rallyhealth/weepickle/v1/Api.scala [hunk 3]
```diff
@@ -110,65 +110,67 @@ trait AttributeTagged extends Api{
...
-          if (delegate == null){
-            throw new Abort("invalid tag for tagged object: " + key, index)
+          val key = keyAttr.asInstanceOf[BufferedValue.Str].value0.toString
+          val reader = taggedReader.findReader(key)
+          if (reader == null){
+            throw new Abort("invalid tag for tagged object: " + key)
           }
-          val ctx2 = delegate.visitObject(-1, -1)
+          // Replaying buffered content requires new path tracking for exceptions thrown by the reader.
+          val delegate = JsonPointerVisitor(reader)
```
Comment (htmldoug) [inline weepickle/src/com/rallyhealth/weepickle/v1/Api.scala]: I realized that we'll also lose index/path information when replaying buffered content over a reader that throws, i.e. higher level parser will always point to the `}`. We can restore path information with `JsonPointerVisitor`. Paths will be relative to the replayed object, but that's good enough. Those will be further wrapped at the parser level. We do lose the exact index within the object, but it's really not worth it.

**Pull request merged**